src/device.rs

```diff
@@ -3,22 +3,23 @@ use std::str;
 use std::ffi::{CStr,OsStr};
 use std::path::Path;
 use std::str::FromStr;
+use std::marker::PhantomData;
 
 use libc::{c_char,dev_t};
 
 use ::context::Context;
 use ::handle::*;
 
-pub fn new(context: &Context, device: *mut ::ffi::udev_device) -> Device {
+pub fn new<'a>(device: *mut ::ffi::udev_device) -> Device<'a> {
```

Comment:
How would the compiler know what lifetime to assign to `'a` without taking any references with `'a`? I think `context` is still needed here, but it can be changed to `PhantomData` like this change in libusb-rs: https://github.com/dcuddeback/libusb-rs/commit/0eaf40f4f1a50913e157cfc429cd6af00a45f094#diff-8f9dfd91ea99cdce77b0fe65d6704a02R200.

Reply:
Type/lifetime inference would be done at the call site.

For example, if I write a function
```rust
fn get_device() -> Device<'static> {
  ::device::new(...)
}
```
would create a new `Device` with a static lifetime.

In this crates' case, in both cases where this function is used, the lifetime can be inferred.

- In `device_from_syspath` I modified the return type to clearly describe the lifetime.
- In `monitor.rs` it can be inferred from the returned `Event`'s lifetime.

Reply:
I see. That makes sense, but I think that's more error-prone than propagating an existing lifetime. It'd be a lot easier for me to maintain if the `PhantomData` is created once based on the context's lifetime and then simply copied to derivative objects.